### ISSUE TITLE
Recursion checks fail when built with MSYS2

### ISSUE THREAD

**Issue by nophead** [openscad/openscad#2005]
Recent Windows snapshots correctly detect recursion but when built with MSYS2 and run on Win7 64bit the openscad.exe terminates with exit code C00000FD, which is stack overflow.

**Comment by t-paul** [comment]
On Windows we currently [assume a fixed stack size](https://github.com/openscad/openscad/blob/master/src/PlatformUtils-win.cc#L100) which is set in [mingw-cross-env.pri](https://github.com/openscad/openscad/blob/master/mingw-cross-env.pri#L38).

**Comment by nophead** [comment]
Looks like that doesn't get passed when doing a MSYS2 build. I.e. that section isn't active in the qmake.

**Comment by nophead** [comment]
I added it to the linker options in a win* section and that fixes the MSYS2 build. I don't see any point in adding it to the QMAKE_CXXFLAGS as it is only the linker that needs to know.

```
win* {
  RC_FILE = openscad_win32.rc
  QMAKE_CXXFLAGS += -DNOGDI
  WINSTACKSIZE = 8388608 # 8MB # github issue 116
  QMAKE_LFLAGS += -Wl,--stack,$$WINSTACKSIZE
}
```

Should I also remove it from mingw-cross-env.pri? I.e. are win* sections active when cross compiling?

Since it needs to match  the value used in STACK_LIMIT_DEFAULT I think that WINSTACKSIZE should be passed as a -D definition from $$WINSTACKSIZE in openscad.pro.

**Comment by kintel** [comment]
@nophead I think they can be moved to `win*` somewhere. `win*` must be defined, also when cross-compiling. Otherwise some of the source files specified in `openscad.pro` wouldn't be compiled.

**Comment by nophead** [comment]
Fixed by #2010
